### PR TITLE
SPMD communicator test filling the gaps

### PR DESCRIPTION
**Pull request by VadimVoevodkin**

### CODE REVIEW

File: cpp/oneapi/dal/test/ccl_communicator.cpp [hunk 1]
```diff
@@ -36,18 +36,36 @@ class ccl_comm_test : public te::policy_fixture {
...
+        }
     }
 
     template <typename T>
     void test_allreduce(T* buffer, std::int64_t count) {
         get_new_comm().allreduce(buffer, buffer, count, spmd::reduce_op::sum).wait();
     }
 
+    template <typename T>
+    void test_allreduce_v(T& value) {
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/test/ccl_communicator.cpp:65]: What does "_v" means in application to allreduce?

File: cpp/oneapi/dal/test/ccl_communicator.cpp [hunk 2]
```diff
@@ -143,16 +190,52 @@ TEST_M(ccl_comm_test, "allreduce") {
...
+//     constexpr std::int64_t count = 1;
+
+//     float buffer[count];
+//     buffer[0] = 1.0f;
+
+//     SECTION("host") {
+//         test_allreduce(buffer, count);
+//     }
+
+//     SECTION("device") {
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/test/ccl_communicator.cpp:215]: Single value methods aren't expected to work on device - please remove "device" section. This test should work on host - please uncomment it.

File: cpp/oneapi/dal/test/ccl_communicator.cpp [hunk 3]
```diff
@@ -143,16 +190,52 @@ TEST_M(ccl_comm_test, "allreduce") {
...
+//         test_allreduce_on_device(buffer, count);
+//     }
+
+//     const std::int64_t rank_count = get_new_comm().get_rank_count();
+//     REQUIRE(buffer[0] == float(rank_count));
+
+// }
+
+// TEST_M(ccl_comm_test, "empty allreduce") {
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/test/ccl_communicator.cpp:224]: Please add a note: whether the test works now or it is in todo list (to be fixed)?

File: cpp/oneapi/dal/test/ccl_communicator.cpp [hunk 4]
```diff
@@ -121,16 +139,45 @@ TEST_M(ccl_comm_test, "bcast") {
...
+//     SECTION("host") {
+//         test_bcast(empty_buf, count);
+//     }
+
+// #ifdef ONEDAL_DATA_PARALLEL
+//     SECTION("device") {
+//         test_bcast_on_device(empty_buf, count);
+//     }
+// #endif
+// }
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/test/ccl_communicator.cpp:166]: Please delete this commented code before merge

File: cpp/oneapi/dal/test/ccl_communicator.cpp [hunk 5]
```diff
@@ -121,16 +139,45 @@ TEST_M(ccl_comm_test, "bcast") {
...
+
+TEST_M(ccl_comm_test, "bcast single value") {
+    float value;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/test/ccl_communicator.cpp:169]:
```suggestion
    float value = 0.0f;
```

Comment (KulikovNikita) [reply cpp/oneapi/dal/test/ccl_communicator.cpp:169]: Please explicitly initialize values.

**Review by KulikovNikita — APPROVED** [review]
Please delete commented sections

**Review by amgrigoriev — APPROVED** [review]

**Pull request merged**